sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/CertificateClientImpl.java

```diff
@@ -1011,19 +1058,41 @@ private Mono<Response<CertificateOperation>> createCertificateWithResponseAsync(
             .doOnError(error -> LOGGER.warning("Failed to create the certificate - {}", certificateName, error));
     }
 
-    public PollerFlux<CertificateOperation, KeyVaultCertificateWithPolicy> getCertificateOperation(String certificateName) {
+    private Response<CertificateOperation> createCertificateWithResponse(String certificateName,
+                                                                         CertificatePolicy certificatePolicy,
+                                                                         boolean enabled,
+                                                                         Map<String, String> tags,
+                                                                         Context context) {
+        CertificateRequestParameters certificateRequestParameters = new CertificateRequestParameters()
+            .certificatePolicy(new CertificatePolicyRequest(certificatePolicy))
+            .certificateAttributes(new CertificateRequestAttributes().enabled(enabled))
+            .tags(tags);
+        context = context == null ? Context.NONE : context;
+        context = enableSyncRestProxy(context);
+
+        return service.createCertificate(vaultUrl, certificateName, apiVersion, ACCEPT_LANGUAGE,
+                certificateRequestParameters, CONTENT_TYPE_HEADER_VALUE, context);
+    }
+
+    public PollerFlux<CertificateOperation, KeyVaultCertificateWithPolicy> getCertificateOperationAsync(String certificateName) {
         return new PollerFlux<>(getDefaultPollingInterval(),
             (pollingContext) -> Mono.empty(),
-            createPollOperation(certificateName),
-            cancelOperation(certificateName),
-            fetchResultOperation(certificateName));
+            createPollOperationAsync(certificateName),
+            cancelOperationAsync(certificateName),
+            fetchResultOperationAsync(certificateName));
+    }
+
+    public SyncPoller<CertificateOperation, KeyVaultCertificateWithPolicy> getCertificateOperation(String certificateName, Context context) {
+        return SyncPoller.createPoller(getDefaultPollingInterval(),
+            (pollingContext) -> new PollResponse<>(LongRunningOperationStatus.NOT_STARTED, null),
+            createPollOperation(certificateName, context),
+            cancelOperation(certificateName, context),
+            fetchResultOperation(certificateName, context));
     }
 
     public Mono<Response<KeyVaultCertificateWithPolicy>> getCertificateWithResponseAsync(String certificateName,
                                                                                          String version,
                                                                                          Context context) {
-        context = context == null ? Context.NONE : context;
```

Comment:
why remove this?

Reply:
I think I accidentally did that, good catch.

Reply:
Oh, I see why I did it. We're always using the `withContext()` method before calling this, so we don't need to null-check the `Context` object in the async versions of methods.